packages/mgt-spfx-utils/README.md

```diff
@@ -0,0 +1,154 @@
+  private _hasImportedMgtScripts = false;
+  private _errorMessage = '';
+
+  protected onInit(): Promise<void> {
+    if (!Providers.globalProvider) {
+      Providers.globalProvider = new SharePointProvider(this.context);
+    }
+    customElementHelper.withDisambiguation('foo');
+    return super.onInit();
+  }
+
+  private onScriptsLoadedSuccessfully() {
+    this.render();
+  }
+
+  public render(): void {
+    importMgtComponentsLibrary(this._hasImportedMgtScripts, this.onScriptsLoadedSuccessfully, this.setErrorMessage);
+
+    this.domElement.innerHTML = `
+    <section class="${styles.helloWorld} ${this.context.sdks.microsoftTeams ? styles.teams : ''}">
+      ${this._renderMgtComponents()}
+      ${this._renderErrorMessage()}
+    </section>`;
+  }
+
+  private _renderMgtComponents(): string {
+    return this._hasImportedMgtScripts
+      ? '<mgt-foo-login></mgt-foo-login>'
+      : '';
+  }
+
+  private setErrorMessage(e?: Error): void {
+    if (e) this.renderError(e);
+
+    this._errorMessage = 'An error ocurred loading MGT scripts';
+    this.render();
+  }
+
+  private _renderErrorMessage(): string {
+    return this._errorMessage
+      ? `<span>${this._errorMessage}</span>`
+      : '';
+  }
+}
+```
+
+### When using React to build webparts
+
+When building SharePoint Framework web parts using React any component that imports from the `@microsoft/mgt-react` library must be asynchronously loaded after configuring the disambiguation setting. The `lazyLoadComponent` helper function exists to facilitate using `React.lazy` and `React.Suspense` to lazy load these components from the top level webpart.
+Below is a minimal example webpart that demonstrates how to use MGT with disambiguation in React based SharePoint Framework Webparts. A complete example is available in the [React SharePoint Webpart Sample](../../samples/sp-webpart/src/webparts/mgtDemo/MgtDemoWebPart.ts).
+
+```ts
+// [...] trimmed for brevity
+import { Providers } from '@microsoft/mgt-element/dist/es6/providers/Providers';
+import { customElementHelper } from '@microsoft/mgt-element/dist/es6/components/customElementHelper';
+import { SharePointProvider } from '@microsoft/mgt-sharepoint-provider/dist/es6/SharePointProvider';
+
+// Async import of component that imports the React Components
+const MgtDemo = React.lazy(() => import('./components/MgtDemo'));
+
+export interface IMgtDemoWebPartProps {
+  description: string;
+}
+// set the disambiguation before initializing any webpart
+customElementHelper.withDisambiguation('bar');
+
+export default class MgtDemoWebPart extends BaseClientSideWebPart<IMgtDemoWebPartProps> {
+  // set the global provider
+  protected async onInit() {
+    if (!Providers.globalProvider) {
+      Providers.globalProvider = new SharePointProvider(this.context);
+    }
+  }
+
+  public render(): void {
+    const element = lazyLoadComponent(MgtDemo, { description: this.properties.description });
```

Comment:
Where is `lazyLoadComponent` coming from?

Reply:
Good catch, thank you!